src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java

```diff
@@ -54,85 +55,85 @@ public class AdvertisementController {
     private final CategoryService categoryService;
     private final LocationService locationService;
 
-    @GetMapping(ApiKey.ADV_THUMBNAIL)
+    @GetMapping(value = ApiKey.ADV_THUMBNAIL, produces = MediaType.APPLICATION_JSON_VALUE)
     @Operation(summary = "Find requested quantity of the advertisement as thumbnails and return them as a page result with filters")
     @ApiResponses(value = {
             @ApiResponse(responseCode = "200", description = "OK"),
             @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
             @ApiResponse(responseCode = "404", description = "NOT FOUND")})
     public Page<AdvertisementTitleDto> findPaginatedAsThumbnails(
```

Comment:
- [ ]  TODO: We need to extract those parameters into a new request entity.

I'd be very grateful if you have the will to do that here.